### PR TITLE
Fix temporary result selection index and google search encodeUriComponent

### PR DESCRIPTION
**Pull request by mfbx9da4**
Hope you like it 🙂

This fixes the functionality though styles have not been fully updated as seen below.
* Padding of temporary result is off
* ~~Inverted selection text color of items when there is a temporary result~~

Perhaps you could fix the styles?

https://user-images.githubusercontent.com/1690659/162817313-c06b809e-66b2-4013-a008-71f8d5307270.mov

### CODE REVIEW

File: src/widgets/search.widget.tsx [hunk 1]
```diff
@@ -152,14 +143,23 @@ export const SearchWidget: FC<Props> = observer(({style}) => {
               !store.ui.query && section.key !== 'favorites'
                 ? store.ui.favoriteItems.length + index
                 : index
+
+            if (item.type === ItemType.TEMPORARY_RESULT) {
```
Comment (ospfranco) [inline src/widgets/search.widget.tsx:147]: This takes care of the UI part but probably not of the enter key handler, you also need to adjust the behavior there (ui.store -> onKeyDown)

**Review by ospfranco — COMMENTED** [review]
Thanks for the PR, it's good. I mean, my code is not so good anyways, since I'm still trying out stuff but it does solve the problem for now :)

I left one comment though, you need to take care of the enter handler, since right now I think it might hardcoded to only react to the temporary result and not further results, once you take care of that I will update the styles

**Comment by ospfranco** [comment]
Ah no, I just saw the video, seems to be working, cool! thanks!

**Comment by ospfranco** [comment]
Screenshot 000196

No... that's kinda broken 😂 I'll take care of it, thanks anyways

**Comment by mfbx9da4** [comment]
hmmm... how do you get into that state? Trying to break it now...

**Comment by mfbx9da4** [comment]
@ospfranco I think it's just an active styles styling issue, think this fixes it https://github.com/ospfranco/sol/pull/35

**Pull request merged**